### PR TITLE
feat: allow for setting `source_data_modified_utc` to utcnow()

### PR DESCRIPTION
**Pull request by niross**
- Specify at the dataset level whether to use xcom date or utcnow when setting the last modified date for a table.
- Cleanup db_tables.py to fix pylint issues

Closes: https://trello.com/c/FyJUmHp1/887-add-last-updated-metadata-to-dataset-pipelines

### Checklist

* [x] Have tests been added to cover any changes?
* [x] Has the [CHANGELOG](https://github.com/uktrade/data-flow/blob/master/CHANGELOG.md) been updated?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

File: dataflow/operators/db_tables.py [hunk 1]
```diff
@@ -433,6 +437,9 @@ def swap_dataset_tables(target_db: str, *tables: sa.Table, **kwargs):
             new_modified_utc = kwargs['task_instance'].xcom_pull(
                 key='source-modified-date-utc'
             )
+            if new_modified_utc is None and use_utc_now_as_source_modified:
+                new_modified_utc = datetime.datetime.utcnow()
```
Comment (samuelhwilliams) [inline dataflow/operators/db_tables.py:441]: If it's not too difficult, I think I'd be tempted to have it be the timestamp from the start (or maybe the end?) of the `fetch-data` task, as that's a bit closer to the truth. I don't think it makes a world of difference, as without pulling it from the source itself it's not really a canonical timestamp, just a "near enough" ... but might still be nice? I assume airflow makes it relatively easy to grab the end timestamp from a task in the same dag run.

File: dataflow/operators/db_tables.py [hunk 2]
```diff
@@ -519,16 +526,16 @@ def poll_scrape_and_load_data(
         ):
             logger.info(
                 "No newer data has been made available for today, "
-                "and it is now after the scheduled end time for polling "
-                f"({pipeline_instance.__class__.daily_end_time_utc}). "
-                "Ending the job and skipping remaining tasks."
+                "and it is now after the scheduled end time for polling (%s). "
+                "Ending the job and skipping remaining tasks.",
+                pipeline_instance.__class__.daily_end_time_utc,
```
Comment (samuelhwilliams) [inline dataflow/operators/db_tables.py:531]: Hmm, are these linting errors not being enforced by `make check`/PR tests? If not we should add them ...

File: tests/unit/operators/test_db_tables.py [hunk 3]
```diff
@@ -369,7 +382,9 @@ def test_branch_on_modified_date(
 
 def test_poll_scrape_and_load_data(mocker):
     class TestPipeline(_FastPollingPipeline):
-        date_checker = lambda: datetime.now()  # noqa
+        def date_checker():
+            return datetime.now()  # noqa
```
Comment (samuelhwilliams) [inline tests/unit/operators/test_db_tables.py:386]: Presume we don't need the `# noqa` any more here.

File: dataflow/operators/db_tables.py (hunk 1 above)
Comment (niross) [reply dataflow/operators/db_tables.py:441]: Yeah makes sense and should be doble

File: dataflow/operators/db_tables.py (hunk 2 above)
Comment (niross) [reply dataflow/operators/db_tables.py:531]: Yes, i'm a bit confused now. `make check` doesn't pick them up but the pre-commit hook does...

Comment (niross) [reply dataflow/operators/db_tables.py:531]: As discussed, i had my own special pre-commit hook that has now been added to the repo proper :-)

File: dataflow/operators/db_tables.py (hunk 1 above)
Comment (niross) [reply dataflow/operators/db_tables.py]: Ok so not as easy as I had hoped but i've updated it to attempt to get the end date of the fetch task but fall back to utcnow() if not found.

Comment (samuelhwilliams) [reply dataflow/operators/db_tables.py]: Nice

**Review by samuelhwilliams — APPROVED** [review]

**Pull request merged**